### DISCUSSION TITLE
how to use `__as_b32` and `__as_ptr_remote_dsmem` in my own cuda code?

### DISCUSSION THREAD

**Discussion by hellboywyh** [NVIDIA/cccl#1260] [Q&A]
I use`__as_b32` and `__as_ptr_remote_dsmem` in my cuda code as follows:
![image](https://github.com/NVIDIA/cccl/assets/33325753/950f7a4d-36e8-4586-9132-2f0b96fe031e)
and I have added the including as follow:
`#include<__cuda/ptx/ptx_helper_functions.h>`
and I compile it with the option `-Icccl/libcudacxx/libcxx/include`
then I got the following error:
![image](https://github.com/NVIDIA/cccl/assets/33325753/fb51da16-82ec-4f57-a64f-d5497c2d1215)
Is there anyone can help to solve this problem?

**Comment by miscco** [comment]
The ptx_helper_functions header is inside an internal header with a reserved identifier. You should never include them on your own.

Rather than that you should include the respective parent header, in this case `<cuda/ptx>`

**Comment by miscco** [comment]
Oh, is it possible that you are on windows and it has issues with path depth / length?

Please see whether it helps if you move cccl to a different folder like C:/src/cccl
